tests/change_detector/change_detector_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package change_detector
```

Comment:
suggestion(non-blocking, minor): Whilst I like the approach, I do dislike that this test will be executed by default with `go test ./...`, I think it might be nicer if we exclude this test by default, perhaps using a build flag.  This would however complicate the github action matrix stuff.

Reply:
Added build tag here and updated Makefile to include `--tags change_detector`.